src/test/java/io/pinecone/PineconeIndexOperationsTest.java

```diff
@@ -0,0 +1,212 @@
+package io.pinecone;
+
+import com.google.gson.Gson;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.exceptions.PineconeValidationException;
+import okhttp3.*;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.openapitools.client.model.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+
+public class PineconeIndexOperationsTest {
+    private static final Gson gson = new Gson();
+
+    @Test
+    public void testDeleteIndex() throws IOException {
+        Call mockCall = mock(Call.class);
+        when(mockCall.execute()).thenReturn(new Response.Builder()
+                .request(new Request.Builder().url("http://localhost").build())
+                .protocol(Protocol.HTTP_1_1)
+                .code(200)
+                .message("OK")
+                .body(ResponseBody.create("Response body", MediaType.parse("text/plain")))
+                .build());
+        OkHttpClient mockClient = mock(OkHttpClient.class);
+        when(mockClient.newCall(any(Request.class))).thenReturn(mockCall);
+
+        Pinecone client = new Pinecone.Client("testAPiKey").withOkHttpClient(mockClient).build();
+        client.deleteIndex("testIndex");
+
+        verify(mockClient, times(1)).newCall(any(Request.class));
+        verify(mockCall, times(1)).execute();
+    }
+
+    @Test
+    public void testCreatePodIndex() throws IOException {
+        String filePath = "src/test/resources/podIndexJsonString.json";
+        String indexJsonStringPod = new String(Files.readAllBytes(Paths.get(filePath)));
+
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("test_name").dimension(3).metric(IndexMetric.COSINE);
+
+        Call mockCall = mock(Call.class);
+        when(mockCall.execute()).thenReturn(new Response.Builder()
+                .request(new Request.Builder().url("http://localhost").build())
+                .protocol(Protocol.HTTP_1_1)
+                .code(201)
+                .message("OK")
+                .body(ResponseBody.create(indexJsonStringPod, MediaType.parse("application/json")))
+                .build());
+
+        OkHttpClient mockClient = mock(OkHttpClient.class);
+        when(mockClient.newCall(any(Request.class))).thenReturn(mockCall);
+        Pinecone client = new Pinecone.Client("testAPiKey").withOkHttpClient(mockClient).build();
+        client.createIndex(createIndexRequest);
+
+        verify(mockClient, times(1)).newCall(any(Request.class));
+        verify(mockCall, times(1)).execute();
```

Comment:
In general I like how clean each of these tests is, even with all the mocking that's going on. Nice work

Reply:
Appreciated, but credit goes to @rohanshah18 I'm mostly cribbing from what was here originally: https://github.com/pinecone-io/pinecone-java-client/blob/main/src/test/java/io/pinecone/PineconeIndexOperationClientTest.java